pkg/daemon/winsvc/winsvc.go

```diff
@@ -28,11 +30,27 @@ func EnsureServiceState(service Service, state svc.State) error {
 	}
 	switch state {
 	case svc.Running:
-		return service.Start()
+		err = service.Start()
+		if err != nil {
+			return err
+		}
 	case svc.Stopped:
 		_, err = service.Control(svc.Stop)
-		return err
+		if err != nil {
+			return err
+		}
 	default:
 		return errors.New("unexpected state request")
 	}
+	// wait for the state change to actually take place
+	return wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
```

Comment:
These intervals and timeouts set for dealing with kubernetes resources. I'd rather not reuse them here. We shouldn't be waiting more than 30 seconds for this to occur, and we should be rechecking much more frequently.

Reply:
As it is, 10 minutes is a very long time to wait for a program to stop.

Reply:
Currently in WMCO, we try every 15 seconds for 5 minutes for services to start, and every 15 seconds for 10 minutes each (wow) for stopping and deleting. Before we reduce this to say, trying every 5 seconds for 30 seconds max, do we remember why we currently have such a long timeout?

Reply:
I don't understand how we are waiting for 10 minutes. Aren't we waiting for 2 minutes and checking every 15 seconds?

Reply:
My question above is regarding the code under review.

Reply:
Ah you're right, theres `Timeout` and `ResourceChangeTimeout`.
It will be a 2 min wait. Thats definitely not as bad, but still a bit long for a program to respond to a stop request.

>Before we reduce this to say, trying every 5 seconds for 30 seconds max, do we remember why we currently have such a long timeout?

I'm not suggesting changing any WMCO timeouts, the existing WMCO timeouts don't have much relevance here. If we want to do that we can and probably should, but outside this PR. I'm strictly talking about the code being introduced.
 
As for the interval, 5 seconds sounds good to me. The time spent waiting adds up when we need to do it for multiple services.

Reply:
I think 2 minutes will be an outlier and it might cut down on us prematurely timing out. WDYT?

Reply:
Or maybe wait at least a minute? We can introduce new constants specifically for services if needed.

Reply:
^ @openshift/openshift-team-windows-containers

Reply:
I think 2 mins should be enough wait, we can increase the timeout if we see instances of the test prematurely timing out.

Reply:
Thanks folks, I'll update with retrying every 5 seconds until a 2 min total timeout